nemo/collections/multimodal/models/text_to_image/stable_diffusion/ldm/ddpm.py

```diff
@@ -1812,6 +1825,8 @@ def training_step(self, dataloader_iter):
             # async grad allreduce is not currently implemented for O1/autocasting mixed precision training
             # so we all-reduce gradients after the pipeline
             self.allreduce_gradients()  # @sangkug we think this is causing memory to blow up (hurts perf)
+        else:
+            raise ValueError("Either distributed_fused_adam or megatron_amp_O2 needs to be set if ddp_overlap is set")
```

Comment:
Is this guard necessary? It make default config of SD train fail, which worked fine on convergence before

Reply:
Without the guard, when 'ddp_overlap' is set ([which is true by default](https://github.com/NVIDIA/NeMo/blob/d656f226ea51b6cf962607ddfa9b9b2bdf44c5b2/nemo/collections/multimodal/models/text_to_image/stable_diffusion/ldm/ddpm.py#L1738)) and an optimizer different than `distributed_fused_adam` or `megatron_amp_O2` is in use (e.g. `megatron_fused_adam`), the training loop saliently skips optimizer step.

I spent lots of time debugging it, so I added the guard.

Reply:
So we should turn the default ddp_overlap to False? I can change the config file then